materials_io/dft.py

```diff
@@ -0,0 +1,95 @@
+from typing import Union, Iterable, Tuple
+from materials_io.base import BaseParser
+from dfttopif import files_to_pif
+from operator import itemgetter
+from glob import glob
+import itertools
+import os
+
+
+# List of files that are known to the VASP parser
+_vasp_file_names = ["outcar", "incar", "chgcar", "wavecar", "wavcar",
+                    "oszicar", "ibzcar", "kpoints", "doscar", "poscar",
+                    "contcar", "vasp_run.xml", "xdatcar"]
+
+
+class DFTParser(BaseParser):
+    """Extract data from Density Functional Theory calculation results
+
+    Uses the `dfttopif <https://github.com/CitrineInformatics/pif-dft>`_ parser to extract
+    metadata from each file
+    """
+
+    def __init__(self, quality_report=False):
+        """Initialize a featurizer
+
+        Args:
+            quality_report (bool): Whether to generate a quality report
+        """
+        self.quality_report = quality_report
+
+    def group(self, paths: Union[str, Iterable[str]], context: dict = None):
+        # Check arguments
+        if isinstance(paths, (str,)):
+            paths = [paths]
+
+        # Clean paths
+        paths = [os.path.abspath(os.path.expanduser(f)) for f in paths]
+
+        # Find all of the files, and attempt to group them
+        files = filter(os.path.isfile, paths)
+        for group in self._group_vasp(files):
+            yield group
+
+        # Recurse into directories
+        for path in filter(os.path.isdir, paths):
+            for group in self.group(glob(os.path.join(path, '*'))):
```

Comment:
Why are we using `glob` over `os.listdir`?

Reply:
`glob` returns the paths with the full path, in contrast to just the filename with `listdir`. I like being able to avoid having to `join` the directory back on again.

Reply:
That's a fair point.